get_sig_pitch.m

```diff
@@ -1,125 +1,197 @@
-function [pitchsig,yanalframelen_out,ystep_out,nframes_out] = ...
-    get_sig_pitch(sig,fs,pitchlimits,yanalframe_ms,yanalstep_ms,yes_verbose)
-% function [pitchsig,[yanalframelen_out],[ystep_out],[nframes_out]] = ...
-%     getwavepitch(sig,fs,pitchlimits,[yanalframe_ms],[yanalstep_ms],[yes_verbose])
-% for example:
-% pitchlimits = [150 300]; % Hz limits for pitch for isubj = 1
-% yanalframe_ms = 30; % milliseconds (the default)
-% yanalstep_ms  = 10; % milliseconds (the default)
+function [pitchsig,pitch_taxis] = ...
+    get_sig_pitch(sig,fs,pitchlimits,yanalframe_ms,yanalstep_ms,yes_verbose, params)
+% Takes in a signal and returns its pitch (f0). 
+%
+% If params.ptrack_method is set to 'praat' (the default), other Praat
+% settings stored in input arg `params` will be used during the Praat call.
+% If params.ptrack_method is not 'praat', the legacy functionality will be
+% used: pitch is calculated within this function.
+%
+% In the Praat method, the `pitchlimits` input argument is igored, and
```

Comment:
```suggestion
% In the Praat method, the `pitchlimits` input argument is ignored, and
```
We can leave things like this now, but when we get to the step where we're changing the GUI, we could consider replacing the 3rd input argument "pitchlimits" with "params" (and then, inside this function, set pitchlimits = params.pitchlimits to be used by the non-praat algorithm). It could be a good opportunity to get rid of the redundancy.